### PR TITLE
[FEATURE] Added table_expectation template and fixed minor details in self check

### PR DESCRIPTION
**Pull request by gilpasternak35**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [ENHANCEMENT], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- add table_expectation template
- fix small self_check SqlAlchemyBatchDataError
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

Thank you for submitting!

### CODE REVIEW

File: examples/expectations/table_expectation_template.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    ):
+        actual_column_count = metrics.get("table.alphabetical_column_name_count")
+
+        return {
+            "success": actual_column_count == 4,
+            "result": {"observed_value": actual_column_count},
+        }
+
+
+if __name__ == "__main__":
```
Comment (alexsherstinsky) [inline examples/expectations/table_expectation_template.py:241]: Does this run?  Should we add a test for it in its present form?..

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
@gilpasternak35 Do we need a test proving that the template will run and produce expected results?

File: examples/expectations/table_expectation_template.py (hunk 1 above)
Comment (gilpasternak35) [reply examples/expectations/table_expectation_template.py:241]: This does - I have a test under the examples section 👍

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Pull request merged**